cmd/gokr-rebuild-kernel/kernel.go

```
@@ -96,7 +96,46 @@ func find(filename string) (string, error) {
 	return "", fmt.Errorf("could not find file %q (looked in . and %s)", filename, path)
 }
 
+func getContainerExecutable() (string, error) {
```

Why not just:

```go
choices := []string{"docker", "podman"}
for _, exe := range choices {
  p, err := exec.LookPath(exe)
  if err != nil {
    continue
  }
  resolved, err := filepath.EvalSymlinks(p)
  if err != nil {
    return "", err
  }
  return resolved, nil
}
return "", fmt.Errorf("none of %v found in $PATH", choices)
```

I think it’s better to not fall back when filepath.EvalSymlinks fails, as it points out a problem in the user’s setup: if the executable is present, but cannot be fully resolved, that should likely be fixed and a transparent fallback might otherwise mask the issue.

Returning a custom error message instead of retaining exec.LookPath’s error is most likely good enough here.

---

No that's better by all means. I was just over thinking what I was writing.